src/HID.cpp

```diff
@@ -160,8 +160,8 @@ bool HID_::setup(USBSetup& setup) {
 
       if (length == sizeof(setReportData)) {
         USB_RecvControl(&setReportData, length);
```

Comment:
I don't fully understand the problem that led to these changes, so please correct me if I'm wrong.  I think the premise is that `USB_RecvControl(void*, int)` in the GD32 core writes only even numbers of bytes, which is why the `setReportData.leds` was changed from a single byte to a two-byte union (so that it wouldn't write out of bounds in the two lines below?).  But in the line above, we're now calling the same function with a pointer to `setReportData` and its size, with is now `3`, so wouldn't this trigger the same problem, causing `USB_RecvControl()` to write four bytes (of a three-byte structure)?  And if that's not a problem because `setup.wLength == 2`, we're not calling `USB_RecvControl()` at all, which seems like it would also be a problem.

Wouldn't it be better to not alter the data structure (especially by employing a widely-misunderstood device like a C++ union), but rather to pass `USB_RecvControl()` a pointer to a temporary local variable, from which we can then extract the desired data for storage?  I think it would make a lot of sense to keep this hack isolated to the interface with the library that's the source of the issue, rather than bringing it up to a higher level, where it's much more confusing.